.github/workflows/tests.yml

```diff
@@ -98,11 +105,28 @@ jobs:
       run: |
         Rename-Item -Path "./dist/$((Get-ChildItem -Path "./dist").Name)" -NewName "Ghidrathon.zip"
         tar -xf ./dist/Ghidrathon.zip -C ../tmp/ghidra/ghidra_PUBLIC/Ghidra/Extensions/
-    - name: Set Ghidrathon Python interpreter
+
+    # test Ghidrathon when GHIDRATHON_SAVE_PATH is unset
+    - name: Set Ghidrathon Python interpreter (default ghidrathon.save path)
+      run: python util/ghidrathon_configure.py ../tmp/ghidra/ghidra_PUBLIC -d
+    - name: Run tests (default ghidathon.save path)
+      run: |
+        ../tmp/ghidra/ghidra_PUBLIC/support/analyzeHeadless ${{ github.workspace }}/../tmp/ghidra test -Import ${{ github.workspace }}/../tmp/ghidra/ghidra_PUBLIC/GPL/DemanglerGnu/os/linux_x86_64/demangler_gnu_v2_24 -overwrite -PostScript ${{ github.workspace }}/data/python/tests/hello.py -PostScript ${{ github.workspace }}/data/python/tests/runall.py > ../tmp/log.txt
+    - name: Check tests (default )
+      run: |
+        python -c "import pathlib, sys;log_text=pathlib.Path('../tmp/log.txt').read_text(encoding='utf-8');print(log_text);sys.exit(0 if 'runall.py called exit with code 0' in log_text else -1)"
+        python -c "import pathlib, sys; sys.exit(0 if pathlib.Path('hello.txt').exists() else -1)"
+
+    # test Ghidrathon when GHIDRATHON_SAVE_PATH is set
+    - name: Set Ghidrathon Python interpreter (GHIDRATHON_SAVE_PATH is set)
+      env:
+        GHIDRATHON_SAVE_PATH: ${{ matrix.ghidrathon-save-path }}
       run: python util/ghidrathon_configure.py ../tmp/ghidra/ghidra_PUBLIC -d
-    - name: Run tests
+    - name: Run tests (GHIDRATHON_SAVE_PATH is set)
+      env:
+        GHIDRATHON_SAVE_PATH: ${{ matrix.ghidrathon-save-path }}
       run: |
-        ../tmp/ghidra/ghidra_PUBLIC/support/analyzeHeadless ${{ github.workspace }}/../tmp/ghidra test -Import ${{ github.workspace }}/../tmp/ghidra/ghidra_PUBLIC/GPL/DemanglerGnu/os/linux_x86_64/demangler_gnu_v2_24 -PostScript ${{ github.workspace }}/data/python/tests/hello.py -PostScript ${{ github.workspace }}/data/python/tests/runall.py > ../tmp/log.txt
+        ../tmp/ghidra/ghidra_PUBLIC/support/analyzeHeadless ${{ github.workspace }}/../tmp/ghidra test -Import ${{ github.workspace }}/../tmp/ghidra/ghidra_PUBLIC/GPL/DemanglerGnu/os/linux_x86_64/demangler_gnu_v2_24 -overwrite  -PostScript ${{ github.workspace }}/data/python/tests/hello.py -PostScript ${{ github.workspace }}/data/python/tests/runall.py > ../tmp/log.txt
```

Comment:
The following in my opinion is a bit compilcated, and could maybe be improved?

Ghidra's `analyzeHeadless` option `-overwrite` is added here to overwrite the imported file `demangler_gnu_v2_24` and avoid conflicting files. This is done because the first step (GHIDRATHON_SAVE_PATH is unset) and the next step (GHIDRATHON_SAVE_PATH is set) would conflict because of the existence of the same `demangler_gnu_v2_24`.

Reply:
I think this is reasonable but I'm open to your suggestions. Otherwise, let's simply add a comment explaining why the `-overwrite` flag must be used.

Reply:
I expiremented a bit with a different appaorch by defining a new variable in matrix strategy but ended up with _too many jobs_, 24 to be specific. I think we can adopt this approach as-is. I added a comment on `-overwrite` flag in f8c3c1d67d1a5238b152c481f7c2271db8d7af00.

Please note, regarding the steps "`GHIDRATHON_SAVE_PATH` is unset", the `-overwrite` flag is unnecessary. It doesn't have any effect in those particular steps. However, the `-overwrite` flag will be required for the subsequent steps to override the existing file.

Reply:
Great, thanks for looking into it and providing a thorough explanation. I wholly support keeping the job count to a minimum 🚀